openmdao/devtools/partition_tree_n2.py

```diff
@@ -60,17 +83,29 @@ def _tree_dict(ss, component_execution_orders, component_execution_index):
 
     return tree
 
-def get_required_data_from_problem(problem):
+def get_required_data_from_problem_or_rootgroup(problem_or_rootgroup):
```

Comment:
this is an oddly named function..  wouldn't `get_model_data` be simpler and more to the point?

I would suggest that the two types for argument would be better explained in a docstring